lib/members_page.rb

```diff
@@ -0,0 +1,72 @@
+# encoding: utf-8
+# frozen_string_literal: true
+require 'scraped'
+require_relative './decorators/party_name_attributes'
+
+class MembersPage < Scraped::HTML
+  decorator Scraped::Response::Decorator::AbsoluteUrls
+  decorator PartyNameAttributes
+
+  field :party_groupings do
+    trs = noko.xpath('//td[contains(.,"Distrito / Circunscripción")]/table/tr')
+    rows = trs.xpath('td[contains(.,"Circunscripción")]/parent::tr')
```

Comment:
This is quite confusing. If this _really_ needs to be so complex, I'd suggest factoring it out to a named method that helps explain what you're targeting. But I'm suspicious of it needing to be.